cmd/circuit/procdkr.go

```diff
@@ -53,6 +59,131 @@ func mkproc(x *cli.Context) {
 	}
 }
 
+func doRun(x *cli.Context, c *client.Client, cmd client.Cmd, path string) {
+
+	w2, _ := parseGlob(path)
+	a2 := c.Walk(w2)
+	_runproc(a2, cmd, x.Bool("tag"))
+
+}
+
+func runproc(x *cli.Context) {
+	defer func() {
+		if r := recover(); r != nil {
+			fatalf("error, likely due to missing server or misspelled anchor: %v", r)
+		}
+	}()
+	c := dial(x)
+	args := x.Args()
+
+	if len(args) != 1 && !x.Bool("all") {
+		fatalf("runproc needs an anchor argument or use the --all flag to do the entire circuit")
+	}
+	buf, _ := ioutil.ReadAll(os.Stdin)
+	var cmd client.Cmd
+	if err := json.Unmarshal(buf, &cmd); err != nil {
+		fatalf("command json not parsing: %v", err)
+	}
+	if x.Bool("scrub") {
+		cmd.Scrub = true
+	}
+
+	el := string("")
+	if len(cmd.Name) > 0 {
+		el = cmd.Name
+	} else {
+		el = cmd.Path
+	}
+
+	if x.Bool("all") {
+
+		w, _ := parseGlob("/")
+
+		anchor := c.Walk(w)
+
+		done := make(chan bool, 10)
+		procs := 0
+
+		for _, a := range anchor.View() {
+
+			procs++
+
+			go func(x *cli.Context, cmd client.Cmd, a string, done chan bool) {
+
+				doRun(x, c, cmd, a)
+				done <- true
+
+			}(x, cmd, a.Path()+"/"+el, done)
+
+		}
+
+		for ; procs > 0 ; procs--  {
+
+			select {
+			case <-done:
+				continue
+			case <-time.After(time.Second * timeout):
+				fmt.Fprintln(os.Stderr, "timeout waiting for the command")
+			}
+
+		}
+
+	} else {
+
+		doRun(x, c, cmd, args[0]+"/"+el)
+
+	}
+
+}
+
+func _runproc(a client.Anchor, cmd client.Cmd, tags bool) {
+
+	p, err := a.MakeProc(cmd)
+	if err != nil {
+		fatalf("mkproc error: %s", err)
+	}
+
+	q := p.Stdin()
+	if err := q.Close(); err != nil {
+		fatalf("error closing stdin: %v", err)
+	}
+
+	if tags {
+
+		done := make(chan bool)
+
+		go func(a string, r io.Reader, w io.Writer, d chan bool) {
+
+			scanner := bufio.NewScanner(r)
+			for scanner.Scan() {
+				fmt.Fprintf(w, "%s %s\n", a, scanner.Text())
+			}
+			if err := scanner.Err(); err != nil {
+				d <- false
+			}
+
+			d <- true
+
+		}(a.Path(), p.Stdout(), os.Stdout, done)
+
+		select {
+		case rv := <-done:
+			if !rv {
+				fmt.Fprintln(os.Stderr, "error prefixing the data")
+			}
+		case <-time.After(time.Second * timeout):
+			fmt.Fprintln(os.Stderr, "timeout waiting for data")
+		}
+
+	} else {
+
+		io.Copy(os.Stdout, p.Stdout())
+
+	}
+
+	a.Scrub()
```

Comment:
a.Scrub() will kill the process immediately. That's a bug. All you need is to set the Scrub parameter in Cmd to true, this will take care of scrubbing the anchor automatically when the process ends. I pointed this out in a comment above.